edi_bank_statement_import/components/process.py

```diff
@@ -0,0 +1,74 @@
+# Copyright 2021 AGEPoly - Téo Goddet
+# @author: Téo Goddet <teo.goddet@gmail.>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import base64
+
+from odoo import _, api
+from odoo.exceptions import UserError
+
+from odoo.addons.component.core import Component
+
+# TODO: add tests
+class EDIExchangeBankStatementInput(Component):
+    """Process bank statements."""
+
+    _name = "edi.input.bank.statement.process"
+    _inherit = "edi.component.input.mixin"
+    _usage = "input.process.bank.statement"
+
+    def __init__(self, work_context):
+        super().__init__(work_context)
+        self.settings = self.type_settings.get("bank_statement", {})
+
+    def process(self):
+        wiz = self._setup_wizard()
+        result = {
+            "statement_ids": [],
+            "notifications": [],
+        }
+
+        file_data = base64.b64decode(wiz.statement_file)
+        wiz.import_single_file(file_data, result)
+
+        if not result['statement_ids']:
+            raise UserError(
+                _(
+                    "You have already imported this file (%s) or this file "
+                    "only contains already imported transactions."
+                ) % wiz.statement_filename
+            )
+        names = []
+
+        if len(result['statement_ids']) == 1:
+            statement = self.env["account.bank.statement"].browse(result['statement_ids'][0])
+            if self._statement_should_be_confirmed():
+                statement.button_post()
+            self.exchange_record.sudo()._set_related_record(statement)
+            return _("Bank Statement %s created") % statement.name
+        else:
+            for statement_id in result['statement_ids']:
+                statement = self.env["account.bank.statement"].browse(statement_id)
+                if self._statement_should_be_confirmed():
+                    statement.button_post()
+                names.append(statement.name)
+    
+                # TODO : I don't know what to do when there is multiple statements, 
```

Comment:
@simahawk 
Here it's possible to have mutiple statements in a single xml file, that mean multiple records for one exchange.
I suppose that it's possible to create child exchange but i could not find infos/example on how it should be done.

Reply:
@TeoGoddet we've discussed this recently. IMO a good approach could be:

1. make exchange_record.exchange_file computed and write-able
2. if the record is a child and there's no value look at the parent

If you take action on this, please do it in a separated PR.

Reply:
#542